### PR TITLE
Rug 0.25.0

### PR DESCRIPTION
**Pull request by Lawouach**
The tests on those are not yet passing. But I have updated so that it compiles and runs with Rug 0.25.1.

### CODE REVIEW

File: .atomist/handlers/event/Deployed.ts [hunk 1]
```diff
@@ -1,35 +1,47 @@
...
+        const image: DockerImage = pod.images[0]
+        const commit: Commit = image.tag.commit
+        const repo: Repo = commit.repo
+        const author = commit.author
+        const person = author.person
+        const chatIdentity = person.chatId
+        const chatName = chatIdentity.screenName
 
-        const cid: string = "commit_event/" + repo.owner() + "/" + repo.name() + "/" + commit.sha()
-
```
Comment (whostolebenfrog) [inline .atomist/handlers/event/Deployed.ts]: These should be lifecycle events surely? Isn't that the whole point of this?

File: .atomist/handlers/event/Deployed.ts [hunk 2]
```diff
@@ -1,35 +1,47 @@
...
         [@state='Started']
-        [/uses::Container()
-            [/isTagged::Tag()
-                [/isTagged::Commit()
-                    [/on::Repo()/channel::ChatChannel()]]]]`
+        [/images::DockerImage()
+            [/tag::Tag()
+                [/commit::Commit()
+                    [/builds::Build()
+                        [/commit::Commit()]
```
Comment (whostolebenfrog) [inline .atomist/handlers/event/Deployed.ts]: Actually I don't think you need this commit either as you are taking it from the one above when constructing the correlation-id (although you've deleted the code in this PR) and thinking more on it - that should be the correct sha to correlate with. I think you can remove it from all of them.

**Review by whostolebenfrog — CHANGES_REQUESTED** [review]
I think all these should be lifecycle messages although I may be misunderstanding something fundamental.

The path expressions look good although as per my comment I think you can remove one more commit from under the build. Giving it more thought it should be fine to use the one on the tag for correlation.

File: .atomist/handlers/event/Deployed.ts (hunk 1 above)
Comment (Lawouach) [reply .atomist/handlers/event/Deployed.ts]: It is indeed. This needs to be updated.

File: .atomist/handlers/event/Deployed.ts (hunk 2 above)
Comment (Lawouach) [reply .atomist/handlers/event/Deployed.ts]: Lol. I'm getting lost :p Would you mind showing me a PXE you think is correct by any chance?

Comment (whostolebenfrog) [reply .atomist/handlers/event/Deployed.ts]: I think what you need is:

```typescript
     `/K8Pod()
        [@state='BackOff']
        [/images::DockerImage()
          [/tag::Tag()
            [/commit::Commit()
              [/builds::Build()
                  [/push::Push()
                      [/commits::Commit()/author::GitHubId()
                          [/person::Person()/chatId::ChatId()]?]
                      [/repo::Repo()]]]]]]`
```

If you start with that we can tweak it if we are missing some data.

Comment (Lawouach) [reply .atomist/handlers/event/Deployed.ts]: Thanks!

**Comment by Lawouach** [comment]
Okay, I'm now using lifecycle message but tests don't pass any longer. I get the following error:

```
├─┬ Container image pulled
    | ├─┬ the handler is triggered: Failed
    | | └── Scenario aborted: Could not evaluate: [com.atomist.rug.InvalidHandlerResultException: Lifecycle messages must contain a GraphNode]
    | └─┬ we should receive a message: Failed
    |   └── Scenario aborted: Could not evaluate: [com.atomist.rug.InvalidHandlerResultException: Lifecycle messages must contain a GraphNode]
```

**Comment by slimslenderslacks** [comment]
I think these latest path expressions look like they'll project out the correct data.  I wrote down the expression that seemed most intuitive to me here:  https://github.com/atomisthq/bot-service/wiki/lifecycle
but I actually think it will probably result in the same data as the one you have.

And @whostolebenfrog is right that these should all be lifecycle messages!

**Pull request merged**